crates/symbolicator-sources/src/remotefile.rs

```diff
@@ -129,7 +133,14 @@ impl Iterator for SourceLocationIter<'_> {
             }
         }
 
-        self.next.pop().map(SourceLocation::new)
+        self.next.pop().and_then(|next| {
+            if !self.previously_returned.contains(&next) {
+                self.previously_returned.insert(next.clone());
+                Some(SourceLocation::new(next))
+            } else {
+                None
+            }
+        })
```

Comment:
I find it a bit weird to maintain a `HashSet` here. Shouldn’t the downstream user make sure that these things are deduplicated?

Reply:
Yeah, that'd be the other way to do it.